scripts/pylib/twister/twisterlib/testplan.py

```diff
@@ -1015,11 +1021,94 @@ def apply_filters(self, **kwargs):
         self.filtered_platforms = set(p.platform.name for p in self.instances.values()
                                       if p.status != "skipped" )
 
+    def _get_required_platform_and_id(self, required_image, instance: TestInstance):
+        # required_image can be in the form of platform:test_id or just test_id
+        split_req_image = required_image.split(':')
+        req_platform = instance.platform.name
+        if len(split_req_image) == 2:
+            req_platform = split_req_image[0]
+        req_test_id = split_req_image[-1]
+        return req_platform, req_test_id
+
+    def _find_required_instance(self, required_image, instance: TestInstance):
+        req_platform, req_test_id = self._get_required_platform_and_id(required_image, instance)
+        for inst in self.instances.values():
+            if req_test_id == inst.testsuite.id and req_platform == inst.platform.name:
+                return inst
+        return None
+
+    def apply_changes_for_required_images(self):
+        # check if required images are in scope
+        for instance in self.instances.values():
+            if not instance.testsuite.required_images:
+                continue
+            if instance.status:
+                # do not proceed if the test is already filtered
+                continue
+
+            if self.options.subset:
+                reason = "Required images are not supported with --subsets"
+                instance.add_filter(reason, Filters.REQUIRED_IMAGE)
+                logger.debug(f"{instance.name}: {reason}")
+                continue
+
+            if self.options.runtime_artifact_cleanup:
+                reason = "Required images are not supported with --runtime-artifact-cleanup"
+                instance.add_filter(reason, Filters.REQUIRED_IMAGE)
+                logger.debug(f"{instance.name}: {reason}")
+                continue
+
+            if instance.testsuite.no_own_image or not Path(
+                    Path(instance.testsuite.source_dir) / 'CMakeLists.txt').exists():
+                instance.no_own_image = True
+                # filter out qemu platforms, because for tests must use origin path to QEMU_PIPE,
```

Comment:
Something is grammatically wrong in this comment, I cannot really comprehend it. Should `for` be here?

Reply:
I removed this comment.